embedded/src/main/java/org/wildfly/core/embedded/EmbeddedProcessFactory.java

```diff
@@ -260,15 +259,14 @@ private static ModuleLoader setupModuleLoader(String modulePath, String... syste
 
         assert modulePath != null : "modulePath not null";
 
-        // verify the supplied modules path exists, and if it does not, stop and allow the user to correct.
+        // verify the the first element of the supplied modules path exists, and if it does not, stop and allow the user to correct.
         // Once modules are initialized and loaded we can't change Module.BOOT_MODULE_LOADER (yet).
 
-        File moduleDir = new File(modulePath);
+        File moduleDir = new File(trimPathToModulesDir(modulePath));
         if (!moduleDir.exists() || !moduleDir.isDirectory()) {
             throw new RuntimeException("The specified module directory " + modulePath + " is invalid or does not exist.");
         }
 
-        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_MODULES_DIR, trimPathToModulesDir(modulePath));
```

Comment:
ServerEnvironment.getModulesDir() is marked deprecated, so until we remove this, we should probably keep this set (just to cover the case where someone is using that method for some reason).

Would you mind updating this to keep the property set? Sorry for the extra work, and thanks a lot for the patch!